ods_ci/tasks/Resources/Provisioning/GPU/amd_operator/amd_operator.sh

```diff
@@ -94,20 +103,66 @@ function machineconfig_updates {
 
 function monitor_logs() {
     local pod_name=$1
-    local search_text=$2
-    local ns=$3
-    local c_name=$4
+    local ns=$2
+    local c_name=$3
+    shift 3
+    local search_text=$(printf "%q " "$@")
     echo "Monitoring logs for pod $pod_name..."
-
     # Use 'kubectl logs' command to fetch logs continuously
-
     oc logs "$pod_name" -c "$c_name" -n "$ns" | while read -r line; do
         if [[ $line == *"$search_text"* ]]; then
             echo "Found \"$search_text\" in pod logs: $line"
         fi
     done
 }
 
+function wait_until_driver_image_is_built() {
+  startup_timeout=$1
+  build_timeout=$2
+  if [[ $? -eq 0 ]];
+  then
+    name=$(oc get pod -n openshift-amd-gpu -l openshift.io/build.name -oname)
+    echo Builder pod name: $name
+    oc wait --timeout="${startup_timeout}s" --for=condition=ready pod -n openshift-amd-gpu -l openshift.io/build.name
+  else 
+    exit 1
```

Comment:
maybe you can add an informative message before `exit 1` to give the user an idea of why it exited

Reply:
I removed the check at all. If the builder pod doesn't get created, we don't reach this step at all